File: intake-corpus-v2/_workspaces/wx-go/validate.go
// Package wxgo holds small, stdlib-only config-field validators (derived from
// lumina vitals/config validation).
package wxgo

import (
	"fmt"
	"strconv"
	"strings"
)

// Sanitize trims surrounding whitespace.
func Sanitize(s string) string {
	return strings.TrimSpace(s)
}

// ParsePositiveInt parses a strictly-positive integer or returns an error.
func ParsePositiveInt(s string) (int, error) {
	v, err := strconv.Atoi(Sanitize(s))
	if err != nil {
		return 0, fmt.Errorf("not an integer: %q", s)
	}
	if v <= 0 {
		return 0, fmt.Errorf("not positive: %d", v)
	}
	return v, nil
}

File: intake-corpus-v2/_workspaces/wx-java/Validate.java
// wx-java: config-field validators (derived from lumina vitals/config
// validation). Default package, no build tool — `javac *.java` + `java <Class>`.
public final class Validate {
    private Validate() {}

    /** Trim surrounding whitespace; reject null. */
    public static String sanitize(String s) {
        if (s == null) {
            throw new IllegalArgumentException("value is null");
        }
        return s.trim();
    }

    /** Parse a strictly-positive integer, or throw. */
    public static int parsePositiveInt(String s) {
        int v = Integer.parseInt(sanitize(s));
        if (v <= 0) {
            throw new IllegalArgumentException("not positive: " + v);
        }
        return v;
    }
}
